lib/membrane_rtc_engine/engine.ex

```diff
@@ -842,115 +805,105 @@ defmodule Membrane.RTC.Engine do
-
-        {format_specific_tee, new_children, filter_links} =
-          cond do
-            format == :raw and filter_tee != nil ->
-              {filter_tee, %{}, []}
-
-            format == :raw and filter_tee == nil and endpoint_tee != nil ->
-              {:tee, endpoint_track_ids} = endpoint_tee
-              prepare_filter_tee(track_id, endpoint_track_ids, state)
-
-            true ->
-              {endpoint_tee, %{}, []}
-          end
-
-        if format_specific_tee do
-          endpoint_name = {:endpoint, endpoint_id}
-
-          new_link =
-            link(format_specific_tee)
-            |> via_out(Pad.ref(:output, endpoint_name))
-            |> via_in(Pad.ref(:input, track_id))
-            |> to({:endpoint, endpoint_id})
-
-          {not_linked,
-           %ParentSpec{
-             children: Map.merge(parent_spec.children, new_children),
-             links: parent_spec.links ++ filter_links ++ [new_link]
-           }}
+    {endpoints_handling_raw_format, endpoints_handling_remote_format} =
+      Enum.split_with(endpoints_to_link, fn endpoint_id ->
+        format = state.subscriptions[endpoint_id][track_id]
+        format == :raw
+      end)
+
+    links =
+      links ++
+        if endpoints_handling_raw_format == [] do
+          []
         else
-          {MapSet.put(not_linked, track_id), parent_spec}
-        end
+          prepare_raw_format_links(track_id, state)
+        end ++
+        Enum.map(endpoints_handling_raw_format, fn endpoint_id ->
+          prepare_track_to_endpoint_links(track_id, endpoint_id, :raw_format_tee)
+        end) ++
+        Enum.map(endpoints_handling_remote_format, fn endpoint_id ->
+          prepare_track_to_endpoint_links(track_id, endpoint_id, :tee)
+        end)
 
-      _track, {not_linked, parent_spec} ->
-        {not_linked, parent_spec}
-    end)
+    state = %{state | waiting_for_linking: waiting_for_linking}
+    {links, state}
   end
 
-  defp prepare_filter_tee(track_id, endpoint_track_ids, state) do
-    filter = {:filter, endpoint_track_ids}
-    filter_tee = {:filter_tee, endpoint_track_ids}
+  defp link_outbound_tracks(tracks_formats, endpoint_id, ctx, state) do
+    {tracks_to_link, tracks_not_to_link} =
+      Enum.split_with(tracks_formats, fn {track_id, _format} ->
+        Map.has_key?(ctx.children, {:tee, track_id})
+      end)
 
-    depayloading_filter = get_in(state, [:filters, track_id])
+    links =
+      Enum.flat_map(tracks_to_link, fn
+        {track_id, :raw} ->
+          if Map.has_key?(ctx.children, {:raw_format_tee, track_id}) do
+            []
+          else
+            prepare_raw_format_links(track_id, state)
+          end ++
+            prepare_track_to_endpoint_links(track_id, endpoint_id, :raw_format_tee)
 
-    children = %{
-      filter => depayloading_filter,
-      filter_tee => Membrane.Tee.PushOutput
-    }
+        {track_id, _remote_format} ->
+          prepare_track_to_endpoint_links(track_id, endpoint_id, :tee)
```

Comment:
We should check if `remote_format` is the same as the format of track. Because this can lead to strange behavior e.g. some endpoint wants to subscribe on track from WebRTCEndpoint but in `:RTMP` format and instead of some error or ignore action he will start to receive RTP packets.